backend/lib/calculate/gravity.py

```diff
@@ -0,0 +1,208 @@
+"""Methods to calculate gravity-based measures of potential spatial accessibility."""
```

Comment:
Some of the functions may be extracted from this file.
Also, might be worth adding some more details / context in the top docstring.